src/link/link.c

```diff
@@ -148,8 +148,7 @@ size_t _z_link_recv_exact_zbuf(const _z_link_t *link, _z_zbuf_t *zbf, size_t len
 
 int8_t _z_link_send_wbuf(const _z_link_t *link, const _z_wbuf_t *wbf) {
     int8_t ret = _Z_RES_OK;
-
-    for (size_t i = 0; (i < _z_wbuf_len_iosli(wbf)) || (ret == -1); i++) {
+    for (size_t i = 0; (i < _z_wbuf_len_iosli(wbf)); i++) {
```

Comment:
if an `iosli` fails to send and a subsequent one succeeds, this function will return `_Z_RES_OK` and it shall not be the case.

Reply:
so here, the behaviour should depend on whether or not the link is streamed:
- on streamed links, keep trying to send what hasn't been sent.
- on datagram links, give up after the first failure, as only fragmented payloads have multi-slice wbufs, which means missing one is a full on transmit failure for the whole thing